### PR TITLE
[Documentation] Moved developer instruction

### PR DESCRIPTION
**Pull request by Akazz**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Moved developer instruction from `dbms/tests/instructions` to the more appropriate path - `docs/{ru|en}/development`

### CODE REVIEW

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Could you please also update ToC and create symlinks from zh, fa languages to en?

**Comment by Akazz** [comment]
> Could you please also update ToC and create symlinks from zh, fa languages to en?

Thanks for pointing this out! I somehow took it for granted that ToC would update itself automatically

File: docs/toc_zh.yml [hunk 1]
```diff
@@ -215,6 +215,7 @@ nav:
   - 'How to Build ClickHouse on Linux for Mac OS X': 'development/build_cross.md'
   - '如何编写C++代码': 'development/style.md'
   - '如何运行ClickHouse测试': 'development/tests.md'
+  - 'The Beginner ClickHouse Developer Instruction': 'development/developer_instruction.md'
```
Comment (zhang2014) [inline docs/toc_zh.yml]: 开发者指南

File: docs/toc_zh.yml [hunk 2]
```diff
@@ -215,6 +215,7 @@ nav:
   - 'How to Build ClickHouse on Linux for Mac OS X': 'development/build_cross.md'
```
Comment (zhang2014) [inline docs/toc_zh.yml]: 如何在Linux中编译Mac OS X ClickHouse

Comment (Akazz) [reply docs/toc_zh.yml]: Thanks, Zhang! Your help with Chinese is always appreciated!

**Pull request merged**